src/uint/boxed.rs

```diff
@@ -248,6 +261,13 @@ impl NonZero<BoxedUint> {
     pub fn widen(&self, bits_precision: u32) -> Self {
         NonZero(self.0.widen(bits_precision))
     }
+
+    /// Creates a new non-zero integer in a const context.
+    /// The second return value is `FALSE` if `n` is zero, `TRUE` otherwise.
+    pub const fn const_new(n: BoxedUint) -> (Self, ConstChoice) {
+        let is_nonzero = n.is_nonzero();
+        (Self(n), is_nonzero)
+    }
```

Comment:
`BoxedUint` can't be instantiated in a `const fn` context because it's heap-allocated

Reply:
When should a function return `(Value, Choice)` instead of `CtChoice(Value)`?

In `boxed/sqrt.rs` the implementation of `sqrt` uses a `NonZero::<BoxedUint>::new(x)` whose value needs to be used whether `x` is zero or not, but `BoxedUint` is not `ConditionallySelectable` so I can't use `CtChoice`'s `map` and `unwrap_or`. It seems like having a `NonZero::<BoxedUint>::new` that returns `(NonZero<BoxedUint>, Choice)` would be necessary, but the current `new`'s signature doesn't work this way, and such a `new` is definitely not `const_new`.

Reply:
The (hopefully temporary) workaround for that sort of thing is [`BoxedUint::conditional_map`](https://docs.rs/crypto-bigint/0.6.0-pre.3/crypto_bigint/struct.BoxedUint.html#method.conditional_map).

I'm [trying to find an upstream solution in `subtle`](https://github.com/dalek-cryptography/subtle/pull/118) but it may be necessary to define traits shaped like the ones in that PR in `crypto-bigint` as a stopgap (and a `CtOption`-alike type, possibly `ConstOption`) /cc @fjarri

Reply:
I saw the `ConstantTimeSelect` trait and will use `ct_select` and `ct_assign` where appropriate. Thank you for the implementation!